tests/core/pyspec/eth2spec/test/altair/sanity/test_blocks.py

```diff
@@ -23,13 +23,19 @@
 def run_sync_committee_sanity_test(spec, state, fraction_full=1.0, rng=Random(454545)):
     all_pubkeys = [v.pubkey for v in state.validators]
     committee = [all_pubkeys.index(pubkey) for pubkey in state.current_sync_committee.pubkeys]
-    participants = rng.sample(committee, int(len(committee) * fraction_full))
+    selected_indices = rng.sample(list(range(len(committee))), int(len(committee) * fraction_full))
+    sync_committee_bits = [True if i in selected_indices else False for i in range(len(committee))]
+    participants = [
+        validator_index
+        for i, validator_index in enumerate(committee)
+        if sync_committee_bits[i]
+    ]
 
     yield 'pre', state
 
     block = build_empty_block_for_next_slot(spec, state)
     block.body.sync_aggregate = spec.SyncAggregate(
-        sync_committee_bits=[index in participants for index in committee],
```

Comment:
#### *Previously*
if `committee = [1, 2, 3, 1]`
`participants = [2, 1]` <— that means the the 2nd and 4th items were selected, not 1st+2nd+4th

`sync_committee_bits=[index in participants for index in committee]` would be return `[True, True, False, True]`

The aggregate signature from `compute_aggregate_sync_committee_signature` would only include the 2nd sig once + 4th sig once rather than the duplicate sigs.

#### This fix
The new fix generates the correct `sync_committee_bits` first and then calculate the corresponding `participants`.